llpc/context/llpcCompiler.cpp

```diff
@@ -876,16 +878,17 @@ Result Compiler::buildPipelineWithRelocatableElf(Context *context, ArrayRef<cons
   bool isUnlinkedPipeline = context->getPipelineContext()->isUnlinked();
   context->getPipelineContext()->setUnlinked(true);
 
-  ElfPackage elf[UnlinkedStageCount];
+  ElfPackage elf[enumCount<UnlinkedShaderStage>()];
   assert(stageCacheAccesses.size() >= shaderInfo.size());
 
   const MetroHash::Hash originalCacheHash = context->getPipelineContext()->getCacheHashCodeWithoutCompact();
   // Print log in the format matching llvm-readelf to simplify testing.
   LLPC_OUTS("LLPC version: " << VersionTuple(LLPC_INTERFACE_MAJOR_VERSION, LLPC_INTERFACE_MINOR_VERSION) << "\n");
   LLPC_OUTS("Hash for pipeline cache lookup: " << formatBytesLittleEndian<uint8_t>(originalCacheHash.bytes) << "\n");
 
-  for (UnlinkedShaderStage stage = static_cast<UnlinkedShaderStage>(0);
-       stage < UnlinkedStageCount && result == Result::Success; ++stage) {
+  for (UnlinkedShaderStage stage : lgc::enumRange<UnlinkedShaderStage>()) {
+    if (result != Result::Success)
+      break;
```

Comment:
I might make more sense to place the break at the end of the loop.  The work is done to compile the shader stage.  At the end, if the result does not succeed, then you break.

Overall, I think this loop body needs to be cleaned up, but that should not be part of this PR.

Reply:
I put this on top to preserve the existing behavior in an obvious way. I moved to the bottom like you suggested, since this may be easier to read.